src/window_identifier/mod.rs

```diff
@@ -310,6 +315,94 @@ impl Serialize for WindowIdentifierType {
     }
 }
 
+impl WindowIdentifierType {
+    /// Sets the given window as a modal child of the window represented by
+    /// self. Currently, setting Wayland window as parent of Wayland window and
+    /// X11 window as parent of X11 window is supported.
+    ///
+    /// This is useful in backend implementations as the portal dialogs have to
+    /// be modal and grouped together with the application that launched the
+    /// request.
+    ///
+    /// Might realize the window if it is not realized yet.
+    ///
+    /// Returns `true` on success.
+    #[cfg(all(
+        any(feature = "gtk4_wayland", feature = "gtk4_x11"),
+        feature = "backend"
+    ))]
+    pub fn set_parent_of(&self, window: &gtk::Window) -> bool {
+        use gtk::prelude::*;
+
+        window.set_modal(true);
+
+        match self {
+            #[cfg(feature = "gtk4_x11")]
+            WindowIdentifierType::X11(xid) => {
+                use gdk4x11::{X11Display, X11Surface};
+
```

Comment:
maybe it is worth trying what libgxdp is using here https://gitlab.gnome.org/GNOME/libgxdp/-/blob/main/src/gxdp-external-window-x11.c?ref_type=heads#L90-105?

Reply:
can you add a comment pointing to where you "copied" the implementation from?

Reply:
Done.